macros.h

```diff
@@ -88,8 +88,13 @@
 
 #define SN_EXPORT_UINT64(name, num) \
   { \
+    if (num < 0) { \
```

Comment:
this check triggers a `-Wtype-limits` warning when building, but shall we keep it as a sanity check?

Reply:
let me check

Reply:
would remove, doesn't make sense since we are the only users of this api